File: client/src/app/modules/quotation/services/quotation.label.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { QuotationLabelType } from '../../quotation/model/QuotationLabelType';

@Injectable({
  providedIn: 'root'
})
export class QuotationLabelTypeService extends AppRestService<QuotationLabelType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getQuotationLabelTypes() {
    return this.getList(new HttpParams(), "quotation-label-types");
  }

  addOrUpdateQuotationLabelType(quotationLabelType: QuotationLabelType) {
    return this.addOrUpdate(new HttpParams(), "quotation-label-type", quotationLabelType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/quotation/service/QuotationLabelTypeService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.QuotationLabelType;

public interface QuotationLabelTypeService {
    public List<QuotationLabelType> getQuotationLabelTypes();

    public QuotationLabelType getQuotationLabelType(Integer id);
}
